source/module_io/output_log.cpp

```diff
@@ -29,6 +29,153 @@ void output_efermi(bool& convergence, double& efermi, std::ofstream& ofs_running
     }
 }
 
+void output_vacuum_level(const ModulePW::PW_Basis* rho_basis,
```

Comment:
It seems that you don't need rho_basis, you just need some int variables from rho_basis. Could you change the interface so that the function doesn't depend on PW_Basis?